src/only.ts

```diff
@@ -0,0 +1,42 @@
+import { IterableContainer } from './_types';
+import { purry } from './purry';
+
+type Only<T extends IterableContainer> = T extends
+  | []
```

Comment:
this should probably be `readonly` too. I think the readonly case works right now but falls down to the last case (where it's handled trivially correctly).

```suggestion
  | readonly []
```